input/postgres/collection.go

```diff
@@ -56,5 +80,69 @@ func NewCollection(ctx context.Context, logger *util.Logger, server *state.Serve
 		Roles:                     roles,
 		ConnectedAsSuperUser:      connectedAsSuperUser,
 		ConnectedAsMonitoringRole: connectedAsMonitoringRole,
+		HelperFunctions:           helpersFromFunctions(helperFunctions),
 	}, nil
 }
+
+func (c *Collection) ForCurrentDatabase(functions []state.PostgresFunction) *Collection {
+	return &Collection{
+		Config:                    c.Config,
+		Logger:                    c.Logger,
+		SelfTest:                  c.SelfTest,
+		GlobalOpts:                c.GlobalOpts,
+		PostgresVersion:           c.PostgresVersion,
+		Roles:                     c.Roles,
+		ConnectedAsSuperUser:      c.ConnectedAsSuperUser,
+		ConnectedAsMonitoringRole: c.ConnectedAsMonitoringRole,
+		HelperFunctions:           helpersFromFunctions(functions),
+	}
+}
+
+func (c *Collection) findHelperFunction(name string, inputTypes []string) (state.PostgresFunction, bool) {
+	funcs, exists := c.HelperFunctions[name]
+	if !exists {
+		return state.PostgresFunction{}, false
+	}
+	for _, f := range funcs {
+		args := strings.Split(f.Arguments, ", ")
+		if len(inputTypes) > len(args) {
+			// We're expecting more arguments than the function has
+			continue
+		}
+		mismatch := false
+		for idx, arg := range args {
+			// Split by this assumed pattern: "NAME TYPE DEFAULT DEFAULT_VALUE"
```

Comment:
```suggestion
			// Split by the assumed output pattern of pg_get_function_arguments: "<name> <type> DEFAULT <default>"
```
(or some other rendering that distinguishes the literal in this list from the other values).